### DISCUSSION TITLE
Ozone 1.4.0 still depend on protobuf-2.5.0

### DISCUSSION THREAD

**Discussion by hchautrung** [apache/ozone#7417] [FAQ]
Hi all,

I am currently building the Ozone version for ARM64 architecture raspberry pi5 and noticed that Ozone 1.4.0 still relies on protobuf-2.5.0, which is quite outdated. Is there any plan to update the protobuf dependency to the latest version?

Thanks,
Trung

**Comment by errose28** [comment]
We will likely be removing the proto 2 dependency as part of the Ozone 2.0 release (HDDS-11516). @jojochuang  is working on updating Ozone's Hadoop dependency to 3.4.1 as part of HDDS-11617, which I believe is a prerequisite to this change.

**Reply by hchautrung** [reply]
Noted with thanks

**Comment by jojochuang** [comment]
We haven't tried raspberry, but for Mac M1 you can follow this doc to patch protobuf 2.5 (steps should be similar for raspberry): https://gist.github.com/smengcl/6a269f44d909ef9245a9ad8c7c6d41e0#file-ozone-m1-md

If you use Maven 3.9.x, I had to do the following steps to copy&rename protoc compiler:

cp /Users/weichiu/.m2/repository/com/google/protobuf/protoc/2.5.0/protoc-2.5.0-osx-aarch_64 /Users/weichiu/.m2/repository/com/google/protobuf/protoc/2.5.0/protoc-2.5.0-osx-aarch_64.exe
cp /Users/weichiu/.m2/repository/com/google/protobuf/protoc/3.7.1/protoc-3.7.1-osx-aarch_64 /Users/weichiu/.m2/repository/com/google/protobuf/protoc/3.7.1/protoc-3.7.1-osx-aarch_64.exe

**Reply by hchautrung** [reply]
I can build multi-arch (linux/amd64 and linux/arm64) with M1 and Maven 3.9.6 without any errors, but with the same Dockerfile If I build with pi5 (ubuntu 24.04 LTS), I almost got the error `/usr/local/go/pkg/tool/linux_amd64/compile: signal: segmentation fault (core dumped)` with architecture linux/amd64, linux/arm64 is ok

FYI: I rename to `protoc-${PROTOBUF_VERSION_ARG}-linux-aarch_64.exe` in Dockfile, PROTOBUF_VERSION_ARG is build argument pass in for instance 2.5.0. Surprisingly build in M1 Macbook are successful for both linux/amd64 and linux/arm64.
